lib/mix/test/mix/cli_test.exs

```diff
@@ -47,9 +47,27 @@ defmodule Mix.CLITest do
       end
       """
 
+      File.write! "lib/hello_quiet.ex", """
+      defmodule Mix.Tasks.MyHelloQuiet do
+        use Mix.Task
+
+        @shortdoc "Says hello"
+
+        def run(_) do
+          Mix.shell.info Mix.Project.get!.hello_world
+        end
+      end
+      """
+
       contents = mix ~w[my_hello]
       assert contents =~ "Hello from MyProject!\n"
       assert contents =~ "Compiled lib/hello.ex\n"
+      assert contents =~ "Compiled lib/hello_quiet.ex\n"
+
+      System.put_env("MIX_QUIET", "1")
```

Comment:
`System.cmd` allows us to pass environment variables directly, we just need to slightly change the helper (maybe allowing environment variables as an optional second argument): https://github.com/elixir-lang/elixir/blob/master/lib/mix/test/mix/cli_test.exs#L130-L134